src/KeyUnit.cpp

```diff
@@ -323,45 +315,38 @@ int KeyUnit::getNewID()
     return ++mMaxID;
 }
 
-QString KeyUnit::getKeyName(int keyCode, int modifierCode)
+QString KeyUnit::getKeyName(const Qt::Key keyCode, const Qt::KeyboardModifiers modifierCode) const
 {
     QString name;
-    /*
-     Qt::NoModifier      0x00000000 No modifier key is pressed.
-     Qt::ShiftModifier   0x02000000 A Shift key on the keyboard is pressed.
-     Qt::ControlModifier 0x04000000 A Ctrl key on the keyboard is pressed.
-     Qt::AltModifier     0x08000000 An Alt key on the keyboard is pressed.
-     Qt::MetaModifier    0x10000000 A Meta key on the keyboard is pressed.
-     Qt::KeypadModifier  0x20000000 A keypad button is pressed.
-     Qt::GroupSwitchModifier 0x40000000 X11 only. A Mode_switch key on the keyboard is pressed.
-    */
-    if (modifierCode == 0x00000000) {
-        name += "no modifiers + ";
-    }
-    if (modifierCode & 0x02000000) {
-        name += "shift + ";
-    }
-    if (modifierCode & 0x04000000) {
-        name += "control + ";
-    }
-    if (modifierCode & 0x08000000) {
-        name += "alt + ";
-    }
-    if (modifierCode & 0x10000000) {
-        name += "meta + ";
-    }
-    if (modifierCode & 0x20000000) {
-        name += "keypad + ";
-    }
-    if (modifierCode & 0x40000000) {
-        name += "groupswitch + ";
+    if (modifierCode == Qt::NoModifier) {
+        name = tr("no modifiers + ",
+                  // Intentional comment to separate arguments
+                  "This text is prepended onto the name of the key in a keybinding when there is "
```

Comment:
I don't think that is enough, because if `no modifier + keyname` is NOT a valid way to put something together in the translators language we need to know about it. Putting together a meaningful text from fragments in a language for the ones that do have modifiers is even less likely to work as I thought it was accepted that that way is likely to be problematic. I would accept that this could be shorter (so I am open to suggestions) but felt a detailed explanation was a starting point that cause someone to speak up if this was going to be a problem in any particular locale. Remember the modifiers themselves have not been marked here for translations - and I think we want to know if they should be...

Reply:
We're working with translators here though, many of whom don't have English as their native language. These are not computer science professionals. Have to keep it simple